src/materialized/bin/materialized.rs

```diff
@@ -288,3 +291,73 @@ to improve both our software and your queries! Please reach out at:
 "
     );
 }
+
+/// Attempts to increase the soft nofile rlimit to the maximum possible value.
+fn adjust_rlimits() {
+    // getrlimit/setrlimit can have surprisingly different behavior across
+    // platforms, even with the rlimit wrapper crate that we use. This function
+    // is chattier than normal at the trace log level in an attempt to ease
+    // debugging of such differences.
+
+    let (soft, hard) = match rlimit::Resource::NOFILE.get() {
+        Ok(limits) => limits,
+        Err(e) => {
+            trace!("unable to read initial nofile rlimit: {}", e);
```

Comment:
I kind of feel like all the logs that happen before bailing out should include a `warn!("unable to be certain that your ulimits are appropriate, see https://github.com/MaterializeInc/materialize/issues/SOME_NUM")` so that people know what's going on?

Reply:
Not every platform even has a meaningful concept of limits, though. (And there's basically no way getrlimit can fail unless the platform doesn't support it.) I have a preference for minimizing warn-level log messages that are not extremely high confidence.